src/vulkan.jl

```diff
@@ -0,0 +1,50 @@
+# define Vulkan types locally, so we don't need to add VulkanCore.jl as dependency
+const VkResult = Cuint
+const VkInstance = Ptr{Void}
+const VkSurfaceKHR = Ptr{Void}
+const VkPhysicalDevice = Ptr{Void}
+# I just define VkAllocationCallbacks as a general Ptr{Void} intead of a struct
+# of Ptr{Void}, anyway Base.cconvert will do the convertion
+const VkAllocationCallbacks = Ptr{Void}
+
+"""
+    VulkanSupported()
+Return whether the Vulkan loader has been found.
+"""
+VulkanSupported() = ccall((:glfwVulkanSupported, lib), Cint, ())
+
+"""
+    GetRequiredInstanceExtensions() -> (count, ptr, A)
+Return a 3-element tuple (count, ptr, A) where `A` is an array of names of Vulkan
+instance extensions required by GLFW for creating Vulkan surfaces for GLFW windows;
+`count` is the number of the extensions(the value of `enabledExtensionCount`);
+`ptr` is the pointer of those strings(the value of `ppEnabledExtensionNames`).
+"""
+function GetRequiredInstanceExtensions()
+    count = Ref{Cuint}(0)
+    ptr = ccall((:glfwGetRequiredInstanceExtensions, lib), Ptr{Ptr{Cchar}}, (Ref{Cuint},), count)
+    return count[], ptr, unsafe_string.(unsafe_wrap(Array, ptr, count[]))
+end
+
+"""
+    CreateWindowSurface(instance, window, allocator=C_NULL) -> surface
+Create a Vulkan surface for the specified window. `surface` is of type `Ref{vk.VkSurfaceKHR}`.
+"""
+function CreateWindowSurface(instance, window::Window, allocator=C_NULL)
+    surface = Ref{VkSurfaceKHR}(C_NULL)
+    ccall((:glfwCreateWindowSurface, lib), VkResult, (VkInstance, WindowHandle, Ptr{VkAllocationCallbacks}, Ptr{VkSurfaceKHR}), instance, window, allocator, surface)
```

Comment:
I believe the fourth input type should be `Ref{VkSurfaceKHR}`.

Reply:
Why not Ptr? I thought it's always `ccall(:func, Void, (Ptr{T},), ::Ref{T})`

Reply:
it seems `Ref{}` is recommended in the [doc](https://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/#Passing-Pointers-for-Modifying-Inputs-1).